### DISCUSSION TITLE
Train with Local Model, diffusers or ckpt?

### DISCUSSION THREAD

**Discussion by smy20011** [smy20011/dreambooth-gui#45] [Polls]
Hey, owner here.

I'm working on local training functionality and find out that the file dialog only support choose either folder or a ckpt file. We cannot have a dialog choose both. I don't want to add more button to the UI which will make it confusing.

I think It's better to limit the local training functionality to one format to make the UI easier to understand. I want to send out a poll to users.

Which local model format do you want to train with, ckpt or diffusers?

**Comment by ZeroHackz** [comment]
Ckpt

**Comment by smy20011** [comment]
Hey, thank you for voting! Local ckpt -> diffusers conversion is implemented in v0.1.9 in the release page.

Feel free to try & report problem!

**Reply by futurevessel** [reply]
I noticed that it no longer generates class images, before (at least in 1.7, I never used 1.8), it would generate the class images for a certain word once and then cache them in the docker image, and if you increased the amount of class images beyond the default by passing --num_class_images=, then it would generate and cache the new amount of class images. Now it never generates any class images at all.

The 'save every steps' seems to work great, I haven't tried other local models yet.

**Reply by futurevessel** [reply]
Training with a local model failed for me, I tried training against v1-5-pruned-emaonly.ckpt, and I got the following error:

Traceback (most recent call last):
  File "/diffusers/scripts/convert_original_stable_diffusion_to_diffusers.py", line 673, in <module>
    checkpoint = torch.load(args.checkpoint_path)
  File "/opt/conda/lib/python3.7/site-packages/torch/serialization.py", line 712, in load
    return _load(opened_zipfile, map_location, pickle_module, **pickle_load_args)
  File "/opt/conda/lib/python3.7/site-packages/torch/serialization.py", line 1049, in _load
    result = unpickler.load()
  File "/opt/conda/lib/python3.7/site-packages/torch/serialization.py", line 1042, in find_class
    return super().find_class(mod_name, name)
ModuleNotFoundError: No module named 'pytorch_lightning'

**Reply by smy20011** [reply]
Just pushed a new docker image, do you want to try it again?

**Reply by futurevessel** [reply]
I'd be happy to, where can I find it ? The current 1.9 in the releases states that it was uploaded 22 hours ago, so it can't be that right ?

**Comment by smy20011** [comment]
You can just rerun it. It execution environment is automatically updated

**Reply by futurevessel** [reply]
Ah, ok, I'm on it then!

**Reply by futurevessel** [reply]
Training against local models seem to work perfectly now, I trained against v1-5-pruned-emaonly.ckpt and wd1-2_sd1-4_merged.ckpt with no problems, and the resulting model files worked as expected.

Still doesn't generate any class images though, at first I thought maybe you had changed the defaults, but --with_prior_preservation --prior_loss_weight=1.0 is still there by default in the 'Training Command', I've tried changing --num_class_images= but there are no class images generated at all.

I don't know if this problem was introduced in 1.9 or in 1.8, since the last version I used before 1.9 was 1.7, and there class image generation worked.

**Reply by futurevessel** [reply]
It's weird, if I choose a  --num_class_images= value that is higher than I've used before for a specific class prompt term, it will start generating class images, or if I use a class prompt I've never used before it will also generate new class images.

So, logically, when it isn't generating new images, it should use a cached version of said files, except I deleted the existing docker image before I re-installed, so if it is using cached versions of class images, where are they stored ? I assumed in the smy20011/dreambooth docker image, but then the cached class image files wouldn't persist if I delete it and re-install that image from scratch.
